### ISSUE TITLE
emoji in tag names is not supported

### ISSUE THREAD

**Issue by giann** [GothenburgBitFactory/taskwarrior#2599]
```shell
$ task add "Test" +😀

$ task
ID  Age    P Project     Tag                       Description                             Urg
230  22s                 \ud83d\ude00 test         Test                                     0.9

$ task +😀 delete
Filter: ( tags _hastag_ 😀 )
No tasks specified.
```

Sometimes the emoji just doesn't show up at all.

**Comment by tbabej** [comment]
Hi @giann,

thanks for reporting! This is actually fixed on our development branch (implemented in #2333 ), and should be released soon:

```
[warrior@tw-2.6.0 ~]$ task add "Test" +😀
Created task 1.
[warrior@tw-2.6.0 ~]$ task

ID Age  Tag Description Urg 
 1 1s   😀  Test         0.8

1 task
```

**Comment by giann** [comment]
Awesome!

**Comment by giann** [comment]
Actually do you mean the 2.6.0 branch? because that's what I was using.

**Comment by tbabej** [comment]
@giann Yep, that's what I mean. I produced the above with a build from 2.6.0. Can you double check what your version is with `task diag`?

**Comment by giann** [comment]
```
task 2.6.0
   Platform: Darwin

Compiler
    Version: Apple LLVM 12.0.5 (clang-1205.0.22.9)
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++17

Build Features
     Commit: c1e811a4f
      CMake: 3.18.1
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.6.16
 Build type: release
```

**Comment by tbabej** [comment]
@giann That should work. What is the top commit in the `libshared` submodule?

```
cd src/libshared
git log
```

**Comment by giann** [comment]
`591fa58`

**Comment by giann** [comment]
I pulled master in src/libshared and now it 's working

**Comment by tbabej** [comment]
Good! I can't reproduce even with `591fa58`, so perhaps this was a build with stale libshared, but I'm glad it's all resolved now :+1:

**Comment by hijackeel** [comment]
I'm using 2.6.2, which is the latest currently available on Gentoo.

**Working:**
`task add foo +🐸`

**Not working:**
`task add bar`
`Created task 123.`
`task 123 modify +🐸`

**After trying both above:**
- `task +🐸` lists `foo` (created with the emoji tag), but not `bar` (modified to have emoji tag).

- Getting the details of `foo` shows this:
`Tags          🐸`

- Getting the details of `bar` shows this:
`Tags          \ud83d\udc38`

Is this fixed in Taskwarrior 3?

**Comment by djmitche** [comment]
It does seem to be fixed in taskwarrior 3, at least by my testing:

```
⸩ task add foo +🐸
TASKRC override: /home/dustin/p/taskwarrior/.taskrc
Created task 3.
⸩ task 3
TASKRC override: /home/dustin/p/taskwarrior/.taskrc

Name          Value                                 
ID            3                                     
Description   foo                                   
Status        Pending                               
Entered       2024-08-26 21:46:19 (3s)
Last modified 2024-08-26 21:46:19 (3s)              
Tags          🐸
Virtual tags  LATEST PENDING READY TAGGED UNBLOCKED 
UUID          3486afe8-99ad-4913-a295-94df94421b15
Urgency       0.8                                   
[tag_🐸       x]

    tags    0.8 *    1 =    0.8
                         ------
                            0.8

⸩ task add bar
TASKRC override: /home/dustin/p/taskwarrior/.taskrc
Created task 4.
⸩ task 4 mod +🐸
Modifying task 4 'bar'.
Modified 1 task.
TASKRC override: /home/dustin/p/taskwarrior/.taskrc
⸩ task 4
TASKRC override: /home/dustin/p/taskwarrior/.taskrc

Name          Value                                 
ID            4                                     
Description   bar                                   
Status        Pending                               
Entered       2024-08-26 21:46:30 (10s)
Last modified 2024-08-26 21:46:39 (1s)              
Tags          🐸
Virtual tags  LATEST PENDING READY TAGGED UNBLOCKED 
UUID          af4ce807-eb09-4bf6-917a-da517acd04ca
Urgency       0.8                                   
[tag_🐸       x]

    tags    0.8 *    1 =    0.8
                         ------
                            0.8
```

I'm surprised that changed in Taskwarrior 3.x -- little, if anything, has changed in the command-line parsing code.

**Comment by tbabej** [comment]
Not sure what the gentoo build process is, but this should be fixed since 2.6.0, see #2333 (referenced above).
